src/plugins/wizard/public/application/components/top_nav.tsx

```diff
@@ -34,27 +35,32 @@ export const TopNav = () => {
   const saveDisabledReason = useCanSave();
   const savedWizardVis = useSavedWizardVis(visualizationIdFromUrl);
   const { selected: indexPattern } = useIndexPatterns();
+  const [config, setConfig] = useState<TopNavMenuData[] | undefined>();
 
-  const config = useMemo(() => {
-    if (!savedWizardVis || !indexPattern) return;
+  useEffect(() => {
```

Comment:
Yea both would work fine functionality-wise. But have to wrap the function within a useEffect() hook because otherwise the browser will give an warning "cannot update during an existing state transition". Discussed with Ashwin and we decided to put the getTopNavConfig in the useEffect() hook to eliminate that warning. 
![image](https://user-images.githubusercontent.com/43937633/193903685-608b467b-8e4b-4680-9911-aecad2c4925c.png)